rocketpy/mathutils/function.py

```diff
@@ -2825,7 +2825,12 @@ def _check_user_input(
 
         # check source for data type
         # if list or ndarray, check for dimensions, interpolation and extrapolation
-        if isinstance(source, (list, np.ndarray)):
+        if isinstance(source, (list, np.ndarray, str, Path)):
+            # Deal with csv or txt
+            if isinstance(source, (str, Path)):
+                # Convert to numpy array
+                source = np.loadtxt(source, delimiter=",", dtype=float)
+
```

Comment:
I understand that this is necessary in order to check the validity of the inputs/outputs, but it is quite strange to read the csv here, throw away the result, and read it again in `set_source`.

Reply:
I would say that loading it twice is really not a good idea. It can be a huge problem if the CSV file is large and takes a while to read.

Furthermore, I would not recommend changing this in this hotfix PR. It doesn't exactly fix a bug.

Reply:
Indeed, in retrospect, it is a bit odd to perform the same operations multiple times in check inputs and while setting the source.

Not in the scope of this hotfix PR, but the check inputs could be made as a way of pre-processing the source so as to only return a `np.array` or a `callable` to the `set_source`.

This would decouple responsabilities a fair bit and could be implemented in various ways, maybe with `functools.singledispatch`.